pandas/io/pytables.py

```diff
@@ -4826,7 +4826,9 @@ def _convert_string_array(data: np.ndarray, encoding: str, errors: str) -> np.nd
     # encode if needed
     if len(data):
         data = (
-            Series(data.ravel()).str.encode(encoding, errors).values.reshape(data.shape)
+            Series(data.ravel())
+            .str.encode(encoding, errors)
+            ._values.reshape(data.shape)
```

Comment:
Similar here 
(and actually doing `_values` might be more confusing, as that can be an EA, which don't have a reshape method)

Reply:
(the more explicit here might be to use `np.asarray(..)` or `to_numpy()` instead of `.values`/`._values`

Reply:
This one: as explained above, I find `_values` more confusing here, because if it would be an EA, the code would actually error 
(to be clear, you don't have to agree with my comment, but you can at least reply to them)

Reply:
> (the more explicit here might be to use np.asarray(..) or to_numpy() instead of .values/._values

That is more explicit about what type we end up with, but is less explicit about whether or not it involves any casting/copying.

> I find _values more confusing here, because if it would be an EA, the code would actually error

I think we can consider two potential failure/failure-adjacent modes: 1) the code is incorrect and we're going to get an error, 2) the code is correct but going to make an unwanted/unneeded copy/cast.

I think our tests (and careful code reviews, thank you) make the first case much less likely than the second.

Reply:
OK, that's a fair point (we will have to live with that for now ;))